pkg/cache/cache.go

```diff
@@ -160,20 +164,40 @@ type Options struct {
 	// unless they have a more specific transform set in ByObject.
 	DefaultTransform toolscache.TransformFunc
 
-	// UnsafeDisableDeepCopy indicates not to deep copy objects during get or
-	// list objects for EVERY object.
+	// DefaultUnsafeDisableDeepCopy is the default for UnsafeDisableDeepCopy
+	// for everything that doesn't specify this.
+	//
 	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
 	// otherwise you will mutate the object in the cache.
 	//
 	// This is a global setting for all objects, and can be overridden by the ByObject setting.
-	UnsafeDisableDeepCopy *bool
+	DefaultUnsafeDisableDeepCopy *bool
 
 	// ByObject restricts the cache's ListWatch to the desired fields per GVK at the specified object.
+	// object, this will fall through to Default* settings.
 	ByObject map[client.Object]ByObject
 }
 
 // ByObject offers more fine-grained control over the cache's ListWatch by object.
 type ByObject struct {
+	// Namespaces maps a namespace name to cache setting. If set, only the
+	// namespaces in this map will be cached.
+	//
+	// Settings in the map value that are unset because either the value as a
+	// whole is nil or because the specific setting is nil will be defaulted.
+	// Use an empty value for the specific setting to prevent that.
+	//
+	// It is possible to have specific Config for just some namespaces
+	// but cache all namespaces by using the AllNamespaces const as the map key.
+	// This wil then include all namespaces that do not have a more specific
+	// setting.
```

Comment:
This should also work via DefaultNamespaces right? (i.e. DefaultNamespaces should also handle AllNamespaces)

Reply:
That section was actually incorrectly copied from the doc. The `AllNamespaces` part is not implemented. I've removed this section.